docs/framework/concepts/yaml_format.md

```diff
@@ -0,0 +1,326 @@
+```
+
+In this example a [Layout.InfiniteScrolling](Layout.InfiniteScrolling) module is added, which needs access to the [Selection.All](Selection.All) module.
+To do this, the selection module defines its identifier `all-articles` using the `id` key.
+Then, the scrolling module `references` the selection using the identifier.
+The `lazy-load` reference name is what the scrolling module uses to access this reference.
+The list of available references names depends on each type of module.
+
+## Styles
+
+App creators will likely need to style specific nodes and, to avoid complex CSS selectors, the easiest way to do it is by adding custom CSS classes.
+The format includes the `styles` key to add one or more CSS classes.
+See the example below:
+
+```yaml
+---
+root:
+  type: Controller.Mesh
+  slots:
+    window:
+      type: Window.Simple
+      slots:
+        content:
+          type: Pager.Simple
+          slots:
+            home-page:
+              shortdef: 'Layout.InfiniteScrolling(expand: true)'
+              references:
+                lazy-load: all-articles
+              slots:
+                content:
+                  shortdef: 'ContentGroup.ContentGroup(expand: true)'
+                  styles:
+                    - ContentGroup--articles
+                  slots:
+                    arrangement:
+                      shortdef: 'Arrangement.List(expand: true)'
+                      slots:
+                        card: 'Card.List(expand: true)'
+                    selection:
+                      type: Selection.All
+                      id: all-articles
+                      slots:
+                        filter: Filter.Articles
+                        order: Order.Sequence
+```
+
+In this example, a `ContentGroup--articles` [BEM-style](https://css-tricks.com/bem-101/) CSS class is added to a node.
+After that, the CSS class can be used from the app's theme to style that specific node.
+
+## Imports
+
+Each app must include its own app description, but there is no need to write it from scratch.
+The framework provides pre-defined apps descriptions or presets, for different kinds of app experiences.
+The format includes the `!import` tag to use these presets.
+See the example below:
+
+```yaml
+---
+!import 'news'
+```
+
+In this example the app includes its own complete copy of the `news` preset, provided by the framework.
+The copy occurs while building the app.
+
+## Overrides
+
+Using a preset can be limiting, as app creators often want to customize things.
+One way to overcome this limitation is to manually copy and modify the preset, but [forking](https://simple.wikipedia.org/wiki/Fork_(software_development)) the preset makes the maintenance of the apps harder.
+Overrides provides a better way of handling this situation, by allowing app creators to customize specific parts of a preset without forking it.
+The format includes 2 different override methods:
+
+1) Variables Overrides.
```

Comment:
nitpick: I like putting this in a sentence (e.g. two different override methods: variable overrides and custom overrides.)